### PR TITLE
FDL source, unit tests and app now live in FirebaseDynamicLinks folder.

### PR DESCRIPTION
**Pull request by dmandar**

### CODE REVIEW

File: FirebaseDynamicLinks/Source/FDLURLComponents/FDLURLComponents+Private.h [hunk 1]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-#import "DynamicLinks/Public/FDLURLComponents.h"
+#import "FDLURLComponents.h"
```
Comment (paulb777) [inline FirebaseDynamicLinks/Source/FDLURLComponents/FDLURLComponents+Private.h]: Please keep full repo-relative paths in imports.   It's necessary for copybara

**Comment by morganchen12** [comment]
Don't forget to update the paths [here](https://github.com/firebase/firebase-ios-sdk/blob/master/Dangerfile#L80) and [here](https://github.com/firebase/firebase-ios-sdk/blob/master/scripts/make_release_notes.py#L35).

File: FirebaseDynamicLinks/Source/FDLURLComponents/FDLURLComponents+Private.h (hunk 1 above)
Comment (dmandar) [reply FirebaseDynamicLinks/Source/FDLURLComponents/FDLURLComponents+Private.h]: Sg. Done.

**Comment by dmandar** [comment]
> Don't forget to update the paths [here](https://github.com/firebase/firebase-ios-sdk/blob/master/Dangerfile#L80) and [here](https://github.com/firebase/firebase-ios-sdk/blob/master/scripts/make_release_notes.py#L35).

sg. Done.

**Pull request merged**